### PR TITLE
Add ability to record a help request call

### PR DESCRIPTION
**Pull request by maysakanoni**
**What**
- Remove `has the callback been completed` and `follow up call required` sections (this what decided how callback list was populated using the `initial_callback_completed` and `callback_required` fields)
- Add ability to log a call (call details section)
- This does not replace how calls are removed from the callback list as the fields (`initial_callback_completed` and `callback_required`) are still updated implicitly thought the selection made in the new call details section
- Add validation to call needed section
**Why**
- This is a requirement for test and trace where call handles need to be able to log a call and view a history of the calls

**Screenshot**
image

**Notes**
-This pr doesn't allow call handles to add test and trace call types as they are currently imported

### CODE REVIEW

**Comment by antonyoneill** [comment]
@maysakanoni can you base this PR on a new branch for the call attempts, we'll put the other call attempt work onto that branch
then merge that branch to master when we're ready to go

File: app/assets/scss/app.scss [hunk 1]
```diff
@@ -182,11 +178,22 @@ textarea[readonly] {
...
-  100% { transform: rotate(360deg); }
+  0% {
+    transform: rotate(0deg);
+  }
+  100% {
+    transform: rotate(360deg);
+  }
+}
+label[for="HelpNeeded-3"] {
+  margin-right: 18px;
```
Comment (kosiakkatrina) [inline app/assets/scss/app.scss:198]: Probably better to assign class to that element here ?

File: controllers/help-requests/help_requests.controller.js [hunk 2]
```diff
@@ -61,16 +63,43 @@ const handleFormErrors = (req, res, errors, path) => {
...
+    }
+}
+const handleCallCreation = async (query) => {
+    try{
+        await HelpRequestCallService.createHelpRequestCall(query)
+        .then(result => {
+            if(result.isError === true){
+                return true
+            }
+        });
```
Comment (kosiakkatrina) [inline controllers/help-requests/help_requests.controller.js:98]: What is this part doing?

Comment (kosiakkatrina) [reply controllers/help-requests/help_requests.controller.js:98]: nevermind 😄

File: models/help-requests/help_request_call.model.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+      const headers = {
+        "Content-Type": "application/json",
+        "Content-Length": helpRequestCalldata.length,
+        "x-api-key": config.help_requests_api_key,
+      };
+
+      await axios
+        .post(
+          config.help_requests_api_url + "/" + id + "/calls",
```
Comment (kosiakkatrina) [inline models/help-requests/help_request_call.model.js:25]: I'd write it as `${config.help_requests_api_url}/${id}/calls` maybe nitpicking tho

**Review by kosiakkatrina — APPROVED** [review]

File: app/assets/scss/app.scss (hunk 1 above)
Comment (maysakanoni) [reply app/assets/scss/app.scss:198]: yeah would be nice, with the lbh templating and rendering multiple elements, not sure if we can?

Comment (kosiakkatrina) [reply app/assets/scss/app.scss:198]: Ah, makes sense

**Pull request merged**